simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/MemoryAllocationExports.java

```diff
@@ -0,0 +1,104 @@
+package io.prometheus.client.hotspot;
+
+import com.sun.management.GarbageCollectionNotificationInfo;
+import com.sun.management.GcInfo;
+import io.prometheus.client.Collector;
+import io.prometheus.client.CounterMetricFamily;
+
+import javax.management.Notification;
+import javax.management.NotificationEmitter;
+import javax.management.NotificationListener;
+import javax.management.openmbean.CompositeData;
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryUsage;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class MemoryAllocationExports extends Collector {
+  private final Map<String, AtomicLong> allocatedMap;
+
+  public MemoryAllocationExports() {
+    this(ManagementFactory.getGarbageCollectorMXBeans(), new ConcurrentHashMap<String, AtomicLong>());
+  }
+
+  // Visible for testing
+  MemoryAllocationExports(List<GarbageCollectorMXBean> garbageCollectorMXBeans, Map<String, AtomicLong> allocatedMap) {
+    this.allocatedMap = allocatedMap;
+    AllocationCountingNotificationListener listener = new AllocationCountingNotificationListener(allocatedMap);
+    for (GarbageCollectorMXBean garbageCollectorMXBean : garbageCollectorMXBeans) {
+      ((NotificationEmitter) garbageCollectorMXBean).addNotificationListener(listener, null, null);
+    }
+  }
+
+  @Override
+  public List<MetricFamilySamples> collect() {
+    List<MetricFamilySamples> sampleFamilies = new ArrayList<MetricFamilySamples>();
+    CounterMetricFamily allocated = new CounterMetricFamily(
```

Comment:
You can use a Counter here, rather than having to track this yourself.

Reply:
Good point! That really simplified `collect`.

I also removed the tests for `collect` since they were now only testing `Counter.collect`.